presto-orc/src/main/java/com/facebook/presto/orc/metadata/statistics/ColumnStatistics.java

```diff
@@ -153,7 +189,7 @@ public boolean equals(Object o)
             return false;
         }
         ColumnStatistics that = (ColumnStatistics) o;
-        return Objects.equals(numberOfValues, that.numberOfValues) &&
+        return Objects.equals(getNumberOfValues(), that.getNumberOfValues()) &&
```

Comment:
I think you need to add `hasNumberOfValues` since this will make `!hasNumberOfValues == 0`

```java
return return Objects.equals(hasNumberOfValues, that.hasNumberOfValues) &&
         Objects.equals(getNumberOfValues(), that.getNumberOfValues()) &&
         Objects.equals(booleanStatistics, that.booleanStatistics) &&
         Objects.equals(integerStatistics, that.integerStatistics) &&
         Objects.equals(doubleStatistics, that.doubleStatistics) &&
```